crypto/hmac/hmac.c

```diff
@@ -217,34 +217,14 @@ int HMAC_CTX_copy(HMAC_CTX *dctx, HMAC_CTX *sctx)
 }
 
 unsigned char *HMAC(const EVP_MD *evp_md, const void *key, int key_len,
```

Comment:
I thought we were going to macro this?

Reply:
I thought so, too, and did so for other symbols like `SHA1` and friends in #14741,
but it turned out that the pattern given yesterday in https://github.com/openssl/openssl/issues/14628#issuecomment-819311206 does not (directly) work for `HMAC` because a macro does not support the special (and not thread-safe) feature that the result is placed in a static array in case `out` is NULL.
And in some way or other we must keep this possibility for backward compatibility
and avoid a clash with the different (and better) handling of  `out == NULL` by `EVP_Q_mac()`: it allocates and uses a buffer of suitable length, which will be returned on success and must be freed by the caller.

The current solution (not using a macro) is the most straighforward. 
Of course, we could rename the implementation in `hmac.c` to, e.g., `HMAC_int` and then
```
# define HMAC(evp_md, key, keylen, data, datalen, out, outlen) \
  HMAC_int(evp_md, key, keylen, data, datalen, out, outlen)
```
but this would just add clutter.